jenkinsfiles/ubuntu.Jenkinsfile

```
@@ -0,0 +1,113 @@
+                }
+                    // Copy genesis.dat file into place in data dir, and rename.
+                    sh '''
+                        mkdir ${DATA_DIR}
+                        cp ${GENESIS_DAT_FILE} ${DATA_DIR}/${ENVIRONMENT}-genesis.dat
+                    '''
+            }
+        }
+        stage('Build node deb') {
+            steps {
+                sh '''
+                    docker build\
+                        --build-arg ubuntu_version=${UBUNTU_VERSION}\
+                        --build-arg build_env_name=${ENVIRONMENT_CAP}\
+                        --build-arg build_env_name_lower=${ENVIRONMENT}\
+                        --build-arg build_genesis_hash=$(cat ${GENESIS_HASH_PATH} | tr -cd "[:alnum:]")\
+                        --build-arg build_collector_backend_url=https://dashboard.${DOMAIN}/nodes/post\
+                        --build-arg build_rpc_server_port=${RPC_PORT}\
+                        --build-arg build_listen_port=${LISTEN_PORT}\
+                        --build-arg build_bootstrap=bootstrap.${DOMAIN}:8888\
+                        -f ./scripts/distribution/ubuntu-packages/deb.Dockerfile\
+                        -t ${ENVIRONMENT}-deb ./scripts/distribution/ubuntu-packages/ --no-cache
+                '''
+            }
+        }
+        stage('Publish') {
+            steps {
+                // Copy out the build artifacts. We create a temporary container and use docker
+                // cp. This makes the output artifacts have correct file permissions (they are
+                // owned by the user who ran the script).
+                sh '''
+                    id=$(docker create ${ENVIRONMENT}-deb)
+                    docker cp $id:/out ${ENVIRONMENT}-build
+                    docker rm $id
+                    aws s3 cp ${ENVIRONMENT}-build/${BUILD_FILE} ${OUTFILE} --grants read=uri=http://acs.amazonaws.com/groups/global/AllUsers
```

In some other scripts we check for existence of the file and fail early if the file with the given name already exists. Would this make sense here as well?

---

Yes it would, added the check.